pulsar/src/main/java/org/openconnectors/PulsarSink.java

```diff
@@ -0,0 +1,72 @@
+package org.openconnectors;
+
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.openconnectors.config.Config;
+import org.openconnectors.connect.ConnectorContext;
+import org.openconnectors.connect.SinkConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+import static org.openconnectors.config.ConfigUtils.verifyExists;
+
+public class PulsarSink implements SinkConnector<byte[]> {
+    private static final Logger LOG = LoggerFactory.getLogger(PulsarSink.class);
+
+    PulsarClient client;
+    Producer producer;
+
+    @Override
+    public CompletableFuture<Void> publish(Collection<byte[]> messages) {
+        try {
+            for (byte[] message : messages) {
+                producer.send(message);
```

Comment:
Should use `sendAsync()` and return a future for the last message in the batch

Reply:
will change